pysmt/solvers/msat.py

```diff
@@ -978,16 +981,27 @@ def walk_implies(self, formula, args, **kwargs):
 
     def walk_times(self, formula, args, **kwargs):
         res = args[0]
-        nl_count = 0 if mathsat.msat_term_is_number(self.msat_env(), res) else 1
         for x in args[1:]:
-            if not mathsat.msat_term_is_number(self.msat_env(), x):
-                nl_count += 1
-            if nl_count >= 2:
-                raise NonLinearError(formula)
-            else:
-                res = mathsat.msat_make_times(self.msat_env(), res, x)
+            res = mathsat.msat_make_times(self.msat_env(), res, x)
         return res
 
+    def walk_pow(self, formula, args, **kwargs):
```

Comment:
This function assumes that the power is integer, is this the case in the general logic or something like `pow(x, 1/2)` is allowed? (I am asking for genuine curiosity and because the pysmt function `Pow` that constructs this term accepts non-integer (but constant) exponents)

Reply:
I think the power node can have integer or rational exponent. 
However, to stay in NRA / NIA we should check that the exponent is a positive integer. 
Maybe it should be done for all solvers that support nonlinear arithmetic in a separate PR.